ports-of-call/span.hpp

```diff
@@ -0,0 +1,91 @@
+#ifndef _PORTS_OF_CALL_SPAN_HPP_
+#define _PORTS_OF_CALL_SPAN_HPP_
+
+#include "ports-of-call/portability.hpp"
+
+#include <cassert>
+#include <cstddef>
+
+namespace PortsOfCall {
+
+// ================================================================================================
+// Heavily simplified relative to std::span, but provides a similar concept in a way
+// that's portable to GPUs.
+
+template <typename DataType>
+class span {
+ private:
+  DataType *ptr_{nullptr};
+  std::size_t size_{0};
+
+ public:
+  // Construct an empty span
+  PORTABLE_FUNCTION constexpr span() = default;
```

Comment:
`PORTABLE_FUNCTION constexpr span() noexcept = default` and for other functions also

Reply:
I believe `PORTABLE_FUNCTION` is *not* required for `defaulted` functions. In fact including the annotation produces warnings.

Reply:
I made both suggested changes.

Reply:
I'm confused---aren't these suggestions contradictory?

Reply:
I read the comments as:
* Add `noexcept`
* Remove `PORTABLE_FUNCTION`